drivers/adc/adc_common.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/adc.h>
+
+int adc_gain_invert(enum adc_gain gain,
+		    s32_t *value)
+{
+	int rv = 0;
+
+	switch (gain) {
+	case ADC_GAIN_1_6:
+		*value *= 6;
+		break;
+	case ADC_GAIN_1_5:
+		*value *= 5;
+		break;
+	case ADC_GAIN_1_4:
+		*value *= 4;
+		break;
+	case ADC_GAIN_1_3:
+		*value *= 3;
+		break;
+	case ADC_GAIN_1_2:
+		*value *= 2;
+		break;
+	case ADC_GAIN_2_3:
+		*value = (3 * *value) / 2;
+		break;
+	case ADC_GAIN_1:
+		break;
+	case ADC_GAIN_2:
+		*value /= 2;
+		break;
+	case ADC_GAIN_3:
+		*value /= 3;
+		break;
+	case ADC_GAIN_4:
+		*value /= 4;
+		break;
+	case ADC_GAIN_8:
+		*value /= 8;
+		break;
+	case ADC_GAIN_16:
+		*value /= 16;
+		break;
+	case ADC_GAIN_32:
+		*value /= 32;
+		break;
+	case ADC_GAIN_64:
+		*value /= 64;
+		break;
+	default:
+		rv = -EINVAL;
```

Comment:
Wouldn't it be better to use an assertion for signaling the invalid gain (it's obviously something wrong with the code that tries to use it, and not some other error condition, like hardware issues etc.) and return the converted value (as @nordic-krch also [suggested](https://github.com/zephyrproject-rtos/zephyr/pull/21606/files#r364614158))? This should simplify the output code.

Reply:
I would not consider it better because I like every function that can fail (even if only when used improperly) to provide an error return code, which isn't possible here since valid return values can be negative.  Consider that a gain could theoretically be read at runtime from a configuration structure.